WalletWasabi.Gui/Controls/WalletExplorer/WalletViewModel.cs

```diff
@@ -82,48 +75,76 @@ public void OpenWallet(bool receiveDominant)
 				Actions.Add(sendTab);
 			}
 
-			var receiveTab = new ReceiveTabViewModel(Wallet);
-			var coinjoinTab = new CoinJoinTabViewModel(Wallet);
-			var historyTab = new HistoryTabViewModel(Wallet);
+			_receiveTab = new ReceiveTabViewModel(Wallet);
+			_coinjoinTab = new CoinJoinTabViewModel(Wallet);
+			_historyTab = new HistoryTabViewModel(Wallet);
 
-			var advancedAction = new WalletAdvancedViewModel(Wallet);
-			var infoTab = new WalletInfoViewModel(Wallet);
-			var buildTab = new BuildTabViewModel(Wallet);
+			var advancedAction = new WalletAdvancedViewModel();
+			_infoTab = new WalletInfoViewModel(Wallet);
+			_buildTab = new BuildTabViewModel(Wallet);
 
-			Actions.Add(receiveTab);
-			Actions.Add(coinjoinTab);
-			Actions.Add(historyTab);
+			Actions.Add(_receiveTab);
+			Actions.Add(_coinjoinTab);
+			Actions.Add(_historyTab);
 
 			Actions.Add(advancedAction);
-			advancedAction.Items.Add(infoTab);
-			advancedAction.Items.Add(buildTab);
+			advancedAction.Items.Add(_infoTab);
+			advancedAction.Items.Add(_buildTab);
+		}
+
+		public Global Global { get; }
+		public KeyManager KeyManager => Wallet.KeyManager;
+		public ReactiveCommand<Unit, Unit> LurkingWifeModeCommand { get; }
+
+		public ObservableCollection<ViewModelBase> Actions
+		{
+			get => _actions;
+			set => this.RaiseAndSetIfChanged(ref _actions, value);
+		}
+
+		private CompositeDisposable Disposables { get; set; }
+
+		public void OpenWallet(bool receiveDominant, bool firstWalletOpened)
+		{
+			IsExpanded = true;
+
+			if (!firstWalletOpened)
```

Comment:
How about asking WalletManager if there are any Opened wallet there? So you can remove this whole firstWalletOpened logic.

Reply:
done.